### ISSUE TITLE
Question: Can I flush Base directly without usb? (bricked wheel base)

### ISSUE THREAD

**Issue by mrkeuz** [ProjectSerotonin/vlfx#2]
Hi very interesting project. 

Recently I bricked my CSL base during trying update firmware from official site. 
So my base do not recognize on computer. `lsusb` in showing nothing.

Can I re-flush my base via `JTAG` or similar way? 

I sent message to Fanatec support, but think they will force me to send the base to them for repairs, which I do not really want (I am in another country, and it will be inconvenient for me to send it back and forth)

Could you advise something? Maybe some forums or info with pcb description and restoring/flush instructions?

**Comment by EIREXE** [comment]
> Hi very interesting project.
> 
> Recently I bricked my CSL base during trying update firmware from official site. So my base do not recognize on computer. `lsusb` in showing nothing.
> 
> Can I re-flush my base via `JTAG` or similar way?
> 
> I sent message to Fanatec support, but think they will force me to send the base to them for repairs, which I do not really want (I am in another country, and it will be inconvenient for me to send it back and forth)
> 
> Could you advise something? Maybe some forums or info with pcb description and restoring/flush instructions?

The manual says the CSL can be set to bootloader mode without a PC; that should allow you to flash it:

by pressing the ON/OFF button “ ” on the front plate of the CSL Elite
Wheel Base for about 8 seconds when the CSL Elite Wheel Base is
switched OFF.

Source: https://fanatec.com/media/pdf/72/c2/86/CSL-E-WB-Manual_1jkpIcIVN6pE9l.pdf

I believe the on board JTAG is disabled

**Comment by mrkeuz** [comment]
Nope. Nothing helped. I've tried 8 second and more, reconnect USB during this process. Nothing helped.

JUST FYI:
I found engineer with pic debugger board, he told that we can flush firmware via pic pins somehow. I've already extracted hex's. Will try. I'll post here if everything success.

PS: But I cannot extract firmware key from generic version. I only success with 402 version of driver. v292 and v439 with no success. 
Seems hash of `exe` not equal with predefined entry in list in `vlfx` utility. 

I've got error like: `Binary not found in DB`

Maybe it needs to describe in docs generic way and description about how to extract decrypt key (maybe new issue need created)?

**Comment by EIREXE** [comment]
@mrkeuz The issue is just that the vlfx binary stores the hash for invidual firmware versions.

To be honest, maybe we should just include the decryption key in the program, it's such a trivial and short key it might be legal for us to publish.
